File: Python/P003.py
'''
The prime factors of 13195 are 5, 7, 13 and 29.

What is the largest prime factor of the number 600851475143 ?
'''


def sieve(x):
    large_prime = 2

    while x > large_prime:
        if x % large_prime == 0:
            x = x / large_prime
            large_prime = 2
        else:
            large_prime += 1
    print "Largest Prime Factor: {0}".format(large_prime)

if __name__ == '__main__':
    sieve(600851475143)

File: Java/src/P003.java
/*
 * What is the largest prime factor of the number 600851475143.
 */

public class P003
{

	public static void main(String[] args)
	{

		int largePrime = 2;
		long largeNum = 6000851475143L;
		
		while (largePrime< largeNum)
		{
			if (largeNum % largePrime == 0)
			{
				largeNum /= largePrime;
			} else {
				largePrime += 1;
			}
		}
		System.out.print(largePrime);
	}
}